internal/report/report.go

```diff
@@ -82,7 +82,11 @@ type Options struct {
 }
 
 // Generate generates a report as directed by the Report.
-func Generate(w io.Writer, rpt *Report, obj plugin.ObjTool) error {
+func Generate(w io.Writer, rpt *Report, obj plugin.ObjTool, ui plugin.UI) error {
+	if rpt.unexpNegSamples {
+		ui.PrintErr("Profile has negative sample values.\nIf using the -base flag to compare profiles, consider using the -diff_base flag instead.")
```

Comment:
Could we check if `-base` flag was actually used and print this warning (or this part of the warning) only in that case?

Reply:
It's possible, but I don't think it's trivial. 

I see a few possible options:
1. Call `report.Generate()` with a boolean indicating if `-base` was used to specify a base profile. This requires also passing that boolean into `interactive()` and `generateReport()` or extending `plugin.Options` with a field for specifying if there is a `-base` profile.
2. Add a boolean to profile.Profile to track if a base profile was specified with the `base`.
3. Start tracking -base somehow in `pprofVariables` (this seems undesirable, because I think specifying a base profile from the interactive mode would fail silently).

Any other ideas or preferences?

(There's also the smaller complication of what to name a boolean indicating the a profile has a "base: profile but not a "diff base" profile)

Reply:
I wonder if it would make more sense to print out this warning in fetchProfiles around where pbase and s.DiffBase are checked. The complication is that at that point the sample index is not known, but we could print a warning if any of the metrics has a negative sample and mention all such metrics in the message. The advantage of placing it there is that the message would be printed only once as opposed to for every report in the interactive mode. What do you think?

Reply:
I'm happy with that solution. 
It may lead to the message printing out when people are using -base to look at cumulative profiles, but I think that'll be okay.

Reply:
That's a good point, we don't want the message to be printed out when someone is doing `pprof -sample_index=alloc_space profile1.pb.gz -base profile2.pb.gz`.